kolibri/core/assets/src/api_resource.js

```diff
@@ -0,0 +1,263 @@
+const logging = require('loglevel');
+const rest = require('rest');
+const mime = require('rest/interceptor/mime');
+
+const client = rest.wrap(mime);
+
+/** Class representing a single API resource object */
+class Model {
+  /**
+   * Create a model instance.
+   * @param {object} data - data to insert into the model at creation time - should include at
+   * least an id for fetching.
```

Comment:
> should include at least an id for fetching

What about newly created ones that we want to save, and have the ID created server-side?